src/bicep/modules/sentinelSolution.bicep

```
@@ -0,0 +1,25 @@
+param workspaceName  string
+param workspaceLocation string
+param tags object = {}
+
+resource workspace 'Microsoft.OperationalInsights/workspaces@2020-08-01' existing = {
+  name: workspaceName
+}
+
+resource sentinelSolution 'Microsoft.OperationsManagement/solutions@2015-11-01-preview'= {
+  name: 'SecurityInsights(${workspaceName})'
+  location: workspaceLocation
+  tags:tags
+  dependsOn:[
+    workspace
+  ]
+  properties: {
+    workspaceResourceId: workspace.id
+  }
+  plan: {
+    name: 'SecurityInsights(${workspaceName})'
+    publisher: 'Microsoft'
+    product: 'OMSGallery/SecurityInsights'
+    promotionCode: ''
+  }
+}
```

Let's reuse the `workspace` resource we pull in as a symbolic reference in this module on L5 and let Bicep manage the dependency. We can then both access the name property of that resource and remove the explicit `dependsOn`:

```suggestion
resource sentinelSolution 'Microsoft.OperationsManagement/solutions@2015-11-01-preview'= {
  name: 'SecurityInsights(${workspace.name})'
  location: workspaceLocation
  tags:tags
  properties: {
    workspaceResourceId: workspace.id
  }
  plan: {
    name: 'SecurityInsights(${workspace.name})'
    publisher: 'Microsoft'
    product: 'OMSGallery/SecurityInsights'
    promotionCode: ''
  }
}
```